### PR TITLE
Fix tests broken by moving Message to dataclass

### PR DESCRIPTION
**Pull request by orsinium**
The changes in #513 broke tests a bit that relied on Message being a namedtuple, see comment https://github.com/Bogdanp/dramatiq/pull/513#issuecomment-1342164116 by @anton-petrov.

1. Add `pipe_target` into `Message.options` inside of `pipeline` in the reverse order, so that the message dump inside `pipe_target` recursively contains `pipe_target` of the next message.
2. Add `Message.__lt__`, so that messages can be sorted. The sorting order is the same as it was for namedtuple.
3. Make sure `Message.args` is a tuple when restoring from a dump.

### CODE REVIEW

File: dramatiq/composition.py [hunk 1]
```diff
@@ -45,8 +52,9 @@ def __init__(self, children, *, broker=None):
             else:
                 messages.append(child.copy())
 
-        for message, next_message in zip(messages, messages[1:]):
-            message.options["pipe_target"] = next_message.asdict()
+        rev = messages[::-1]
```
Comment (Bogdanp) [inline dramatiq/composition.py:55]: Is this change necessary?

Comment (orsinium) [reply dramatiq/composition.py:55]: Yes, it is. `asdict` returns a copy of the message content. If you set `pipe_target` going from first to last, the message inside of it won't have its own `pipe_target`. You can try it yourself: remove this line and run the tests, there is one test (I don't remember the name) that reproduces it.

Comment (Bogdanp) [reply dramatiq/composition.py:55]: I see, thanks! So that means the earlier change to `asdict` was technically a change in behaviour. I prefer this new behaviour, but I'll have to think about whether or not it might cause issues downstream.

Comment (orsinium) [reply dramatiq/composition.py:55]: It happens because dataclasses.asdict creates a copy of collections:

image

I think we can preserve the old behavior if we explicitly put the real value of `options` instead of the copy on top of the copied result.

Comment (Bogdanp) [reply dramatiq/composition.py:55]: > I think we can preserve the old behavior if we explicitly put the real value of options instead of the copy on top of the copied result.

Yes, I think we should do that.

Comment (orsinium) [reply dramatiq/composition.py]: Done. I made `Message.asdict` not to create copies and removed reverse loop from `pipeline.__init__`

**Comment by Bogdanp** [comment]
Thanks!
